### ISSUE TITLE
Missing backwards compatibility for `gitlab-ci` section after refactor

### ISSUE THREAD

**Issue by kwryankrattiger** [LLNL/spack#36409]
### Steps to reproduce

Using environment config `spack.yaml`

```yaml
spack:
  specs: [zlib]
  gitlab-ci:
    mappings:
    - match:
      - '@:'
      runner-attributes:
        tags: [donotcare]
```

```console
$ spack env create test spack.yaml
```

### Error message

```console
==> Error: testing/spack.yaml:2: Additional properties are not allowed ('gitlab-ci' was unexpected)
```

### Information on your system

* **Spack:** 0.20.0.dev0 (d15fe6a3459727187dd23b9b18dad18f5fd734fa)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-skylake
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by kwryankrattiger** [comment]
@alalazo @scottwittenburg

**Comment by kwryankrattiger** [comment]
Introduced in #34272
Fixed in #36045

**Comment by alalazo** [comment]
Fixed in https://github.com/spack/spack/pull/36045
